app/src/main/java/org/dhis2/data/prefs/PreferenceProviderImpl.kt

```diff
@@ -56,7 +80,8 @@ class PreferenceProviderImpl(val context: Context) : PreferenceProvider {
     }
 
     override fun areCredentialsSet(): Boolean {
-        return SecurePreferences.getBooleanValue(context, Constants.SECURE_CREDENTIALS, false)
+        return SecurePreferences.getBooleanValue(context, Constants.SECURE_CREDENTIALS,
```

Comment:
for the future (not now), this class should only contains atomic operations not validations or any others